src/qt/buynamespage.cpp

```diff
@@ -78,10 +101,46 @@ void BuyNamesPage::onRegisterNameAction()
     ui->registerNameButton->setDefault(true);
 }
 
+// Returns empty string if available, otherwise a description of why it is not
+// available.
+QString BuyNamesPage::name_available(const QString &name) const
+{
+    std::string strName = name.toStdString();
+    LogPrint(BCLog::QT, "wallet attempting name_show: name=%s\n", strName);
+
+    UniValue params(UniValue::VOBJ);
+    params.pushKV ("name", strName);
+
+    std::string walletURI = "/wallet/" + walletModel->getWalletName().toStdString();
+
+    UniValue res;
+    try
+    {
+        walletModel->node().executeRpc("name_show", params, walletURI);
```

Comment:
I don't remember all details of how the `walletModel` works, but do we need an explicit wallet URI in the RPC call?  The method called itself does not need the wallet, so perhaps it would be best to call it without wallet?  Or does the usage of `walletModel` require it somehow?

Reply:
IIRC omitting the wallet URI results in the default wallet being used instead; it doesn't result in no wallet being used.  Using the default wallet in this kind of code has been a source of bugs for us in the past.  Also, I intend to (in a follow-up PR) make this section of code check the `ismine` field so that the error message can be different if the user tries to register a name that they already own (Electrum-NMC already does this).  So, for these reasons, I prefer not to remove the wallet URI.

Reply:
Sounds good, thanks for the explanation.